express/properties/workflow.py

```diff
@@ -167,17 +167,28 @@ def _construct_predict_subworkflows(self, train_subworkflows: list) -> list:
             for unit in filter(lambda i: "tags" in i, subworkflow["units"]):
                 tags = unit["tags"]
 
+                # Note: This should be the *first* thing we check for, in case we have tags that add properties.
+                # Remove properties if needed
+                if "remove-all-results" in tags:
+                    unit["results"] = []
+
+                # The following tags can be placed in any order:
+
                 # Set predict status
                 if "pyml:workflow-type-setter" in tags:
                     unit["value"] = "True"
 
                 # Set download-from-object-storage units
-                elif 'set-io-unit-filenames' in tags:
+                if 'set-io-unit-filenames' in tags:
                     self.set_io_unit_filenames(unit)
 
-                # Remove properties if needed
-                if "remove-all-results" in tags:
-                    unit["results"] = []
+                # Set predictors to print their predictions to the results tab during the predict phase
+                if 'creates-predictions-csv-during-predict-phase' in tags:
+                    unit["results"] = [{
+                        "name": "file_content",
+                        "basename": "predictions.csv",  # todo: We shouldn't be hardcoding this in to the flavors
```

Comment:
This should be ticketed up to address, if it needs to be addressed, or else it'll never get done.

Reply:
Created - https://exabyte.atlassian.net/browse/SOF-4403